test/util.js

```diff
@@ -513,3 +515,261 @@ function checkCursorPaginatedHistoryObject ({
+ *   and test each one of those values
+ *   otherwise, will perform a fetch of objects and test with fixtures values
+ * @param {Boolean}  [params.filters[i].noResultsExistenceCheck = false]
+ *   by default, will check if there are results while it should have (based on auto-fetched values)
+ */
+async function checkFilters ({
+  t,
+  authorizationHeaders,
+  endpointUrl,
+  fetchEndpointUrl,
+  checkPaginationObject,
+  filters,
+}) {
+  const getEndpoint = (endpointUrl, queryParams) => {
+    if (endpointUrl.includes('?')) return `${endpointUrl}&${queryParams}`
+    else return `${endpointUrl}?${queryParams}`
+  }
+
+  const getRandomValue = (values) => _.first(_.shuffle(values))
+
+  const sortValues = (values) => {
+    return values.slice(0).sort((a, b) => {
+      if (a < b) return -1
+      else if (a > b) return 1
+      else return 0
+    })
+  }
+
+  // prop=value
+  const getExactValueFilter = (prop, exactValue) => `${prop}=${exactValue}`
+
+  // prop[]=value1&prop[]=value2
+  const getArrayValuesFilter = (prop, values) => values.map(v => `${prop}[]=${v}`).join('&')
+
+  // prop=value1,value2
+  const getArrayCommaSeparatedValuesFilter = (prop, values) => `${prop}=${values.join(',')}`
+
+  // prop[gte]=min&prop[lte]=max
+  const getRangeFilter = (prop, range) => `${prop}[gte]=${range.gte}&${prop}[lte]=${range.lte}`
+
+  const addCheckObjFunctions = (filter) => {
+    const {
+      prop,
+      customGetValue,
+      isArrayFilter,
+      isRangeFilter,
+      customCheck
+    } = filter
+
+    const getValue = (obj) => _.isFunction(customGetValue) ? customGetValue(obj) : obj[prop]
+    filter.getValue = getValue
+
+    filter.getExactValueCheckObj = _.curry((exactValue, obj) => {
+      return _.isFunction(customCheck)
+        ? customCheck(obj, exactValue)
+        : getValue(obj) === exactValue
+    })
+
+    if (isArrayFilter) {
+      filter.getArrayValuesCheckObj = _.curry((values, obj) => {
+        return _.isFunction(customCheck)
+          ? customCheck(obj, values)
+          : values.includes(getValue(obj))
+      })
+    } else if (isRangeFilter) {
+      filter.getRangeValuesCheckObj = _.curry((rangeValues, obj) => {
+        return _.isFunction(customCheck)
+          ? customCheck(obj, rangeValues)
+          : rangeValues.gte <= getValue(obj) && getValue(obj) <= rangeValues.lte
+      })
+    }
+  }
+
+  const addTestValues = (filter) => {
+    const {
+      isArrayFilter,
+      isRangeFilter,
+      customTestValues,
+    } = filter
+
+    addCheckObjFunctions(filter)
+
+    const getValue = filter.getValue
+
+    const useCustomValues = customTestValues && customTestValues.length
+
+    const realValues = _.compact(_.uniq(results.map(getValue)))
+    let testValues = useCustomValues
+      ? customTestValues
+      : _.shuffle(realValues).slice(0, 2) // only take few values
+
+    if (!testValues.length) testValues = ['value1', 'value2']
```

Comment:
Why is this needed?

It’s confusing and error-prone since it can make the test useless.

Reply:
Those fake values was to ensure the array filter format is allowed, in the case there aren't any values retrieved from fixtures.
This is indeed confusing, I'll remove it.